File: frontend/src/services/documentsService.ts
import { apiClient } from './api';
import { Document } from '../types/document';

export const documentsService = {
  async getCaseDocuments(caseId: string): Promise<Document[]> {
    const response = await apiClient.get<Document[]>(`/api/documents/cases/${caseId}/documents`);
    return response.data;
  },

  async getDocumentById(documentId: string): Promise<Document> {
    const response = await apiClient.get<Document>(`/api/documents/${documentId}`);
    return response.data;
  },



  async getAllDocuments(): Promise<Document[]> {
    // Get all documents across all cases
    const cases = ['case-001', 'case-002', 'case-003', 'case-004', 'case-005', 'case-006'];
    const allDocuments: Document[] = [];

    for (const caseId of cases) {
      try {
        const caseDocuments = await this.getCaseDocuments(caseId);
        allDocuments.push(...caseDocuments);
      } catch (error) {
        console.warn(`Failed to fetch documents for ${caseId}:`, error);
      }
    }

    return allDocuments;
  },



  async getDocumentContent(documentId: string): Promise<{ document_id: string; content: string; content_length: number }> {
    const response = await apiClient.get<{ document_id: string; content: string; content_length: number }>(`/api/documents/${documentId}/content`);
    return response.data;
  },
};

File: backend/app/services/documents_service.py
#!/usr/bin/env python3
"""
DocumentsService - Document management service for the Legal AI System

This service handles document-related operations including:
- Loading case documents from JSON files
- Document content retrieval
- Document search and filtering
"""

import json
from pathlib import Path
from typing import Dict, List, Any, Optional


class DocumentsService:
    """Service for document management operations."""
    
    @staticmethod
    def load_case_documents(case_id: str) -> List[Dict[str, Any]]:
        """Load all documents for a specific case."""
        try:
            backend_dir = Path(__file__).parent.parent.parent
            case_docs_path = backend_dir / "data" / "cases" / "case_documents" / "case_documents_index.json"
            
            if not case_docs_path.exists():
                return []
            
            with open(case_docs_path, 'r', encoding='utf-8') as f:
                data = json.load(f)
                
                # Handle both flat array and nested structure
                if isinstance(data, dict) and 'case_documents' in data:
                    all_docs = data['case_documents']
                else:
                    all_docs = data
                
                # Filter documents for the specific case
                case_docs = [doc for doc in all_docs if doc.get('case_id') == case_id]
                return case_docs
        except Exception as e:
            print(f"Error loading case documents: {e}")
            return []
    
    @staticmethod
    def load_document_content(document_id: str) -> Optional[str]:
        """Load the full text content of a document."""
        try:
            backend_dir = Path(__file__).parent.parent.parent
            
            # First, find the document in the index to get its path
            case_docs_path = backend_dir / "data" / "cases" / "case_documents" / "case_documents_index.json"
            
            if not case_docs_path.exists():
                return None
            
            with open(case_docs_path, 'r', encoding='utf-8') as f:
                data = json.load(f)
            
            # Handle both flat array and nested structure
            if isinstance(data, dict) and 'case_documents' in data:
                all_docs = data['case_documents']
            else:
                all_docs = data
            
            # Find the document
            document = next((doc for doc in all_docs if doc.get('id') == document_id), None)
            if not document:
                return None
            
            # Load content from the document's file path
            if 'full_content_path' in document:
                content_path = backend_dir / "data" / document['full_content_path']
                if content_path.exists():
                    with open(content_path, 'r', encoding='utf-8') as f:
                        return f.read()
            
            # Fallback to content preview if full content not available
            return document.get('content_preview', '')
            
        except Exception as e:
            print(f"Error loading document content: {e}")
            return None
    
    @staticmethod
    def find_document_by_id(document_id: str) -> Optional[Dict[str, Any]]:
        """Find a document by ID across all cases."""
        try:
            backend_dir = Path(__file__).parent.parent.parent
            case_docs_path = backend_dir / "data" / "cases" / "case_documents" / "case_documents_index.json"
            
            if not case_docs_path.exists():
                return None
            
            with open(case_docs_path, 'r', encoding='utf-8') as f:
                data = json.load(f)
            
            # Handle both flat array and nested structure
            if isinstance(data, dict) and 'case_documents' in data:
                all_docs = data['case_documents']
            else:
                all_docs = data
            
            # Find the document
            document = next((doc for doc in all_docs if doc.get('id') == document_id), None)
            return document
            
        except Exception as e:
            print(f"Error finding document: {e}")
            return None
# 
Export the service class
__all__ = ['DocumentsService']